private/pkg/transport/http/httpserver/zap.go

```diff
@@ -15,13 +15,40 @@
 package httpserver
 
 import (
+	"fmt"
 	"net/http"
 	"time"
 
 	"github.com/go-chi/chi/v5/middleware"
 	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 )
 
+// includes fields described in https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#httprequest
+type httpRequestLog struct {
+	requestMethod string
+	requestURL    string
+	status        int
+	responseSize  string
+	userAgent     string
+	remoteIP      string
+	latency       string
+	protocol      string
```

Comment:
is there any reason why everything here are strings, 
can't they be their native types (responseSize as int etc) and then zap just logs it as its native type?

Reply:
Actually I was following the format of [google cloud's log entry](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#httprequest). Also, I don't think we're going to do this update in the buf repo, and I should close this PR, see issue BSR-431. Thanks for the review though.